tests/monotouch-test/HttpClient/HttpClientTest.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+#if XAMCORE_2_0
+using Foundation;
+#else
+using MonoTouch.Foundation;
+#endif
+using NUnit.Framework;
+
+
+namespace MonoTouchFixtures.HttpClientTests
+{
+	[TestFixture]
+	public class HttpClientTest
+	{
+		const int WaitTimeout = 5000;
+
+		[Test]
+		public void CFNetworkEnsureModifiabilityPostSend ()
+		{
+			var handler = new CFNetworkHandler ();
+			using (var client = new HttpClient (handler))
+			using (var request = new HttpRequestMessage (HttpMethod.Get, "http://xamarin.com")) {
+				Assert.DoesNotThrow (() => handler.AllowAutoRedirect = !handler.AllowAutoRedirect);
+				Task.Factory.StartNew (() => {
+					var token = new CancellationTokenSource ();
+					client.SendAsync (request, token.Token).Wait (WaitTimeout);
+					Assert.Throws<InvalidOperationException> (() => handler.AllowAutoRedirect = !handler.AllowAutoRedirect);
+					// cancel to ensure that we do not have side effects
+					token.Cancel ();
+				});
+			}
+		}
```

Comment:
as discussed (1:1) we should avoid code duplication by using extension methods to expose properties like `AllowAutoRedirect` that exists in many handlers (but not in the base class).